tests/abi_roundtrip.py

```diff
@@ -120,11 +120,12 @@ def array_comp_factory(self) -> pt.ABIReturnSubroutine:
         When the length has not been provided for a dynamic array,
         default to DEFAULT_DYNAMIC_ARRAY_LENGTH
         """
-        if self.length is not None:
-            assert self.type_spec.is_length_dynamic()  # type: ignore[attr-defined]
-        elif not self.type_spec.is_length_dynamic():  # type: ignore[attr-defined]
-            self.length = self.type_spec.length_static()  # type: ignore[attr-defined]
-        else:
+        if not self.type_spec.is_length_dynamic():  # type: ignore[attr-defined]
+            if self.length:
```

Comment:
prefer `is not None` here, since `length` _could_ be 0

Reply:
👍